src/ds5/ds5-private.cpp

```diff
@@ -39,26 +39,71 @@ namespace librealsense
                 << intrinsics_string(res_1280_800)
                 << intrinsics_string(res_960_540));
 
-            auto resolution = width_height_to_ds5_rect_resolutions(width ,height);
-            rs2_intrinsics intrinsics;
-            intrinsics.width = resolutions_list[resolution].x;
-            intrinsics.height = resolutions_list[resolution].y;
+            auto resolution = width_height_to_ds5_rect_resolutions(width, height);
 
-            auto rect_params = static_cast<const float4>(table->rect_params[resolution]);
-            // DS5U - assume ideal intrinsic params
-            if ((rect_params.x == rect_params.y) && (rect_params.z == rect_params.w))
+            if (width == 848 && height == 100) // Special 848x100 resolution is available in some firmware versions
+                // for this resolution we use the same base-intrinsics as for 848x480 and adjust them later
             {
-                rect_params.x = rect_params.y = intrinsics.width * 1.5f;
-                rect_params.z = intrinsics.width * 0.5f;
-                rect_params.w = intrinsics.height * 0.5f;
+                resolution = width_height_to_ds5_rect_resolutions(width, 480);
+            }
+
+            if (resolution < max_ds5_rect_resolutions)
+            {
+                rs2_intrinsics intrinsics;
+                intrinsics.width = resolutions_list[resolution].x;
+                intrinsics.height = resolutions_list[resolution].y;
+
+                auto rect_params = static_cast<const float4>(table->rect_params[resolution]);
+                // DS5U - assume ideal intrinsic params
+                if ((rect_params.x == rect_params.y) && (rect_params.z == rect_params.w))
+                {
+                    rect_params.x = rect_params.y = intrinsics.width * 1.5f;
+                    rect_params.z = intrinsics.width * 0.5f;
+                    rect_params.w = intrinsics.height * 0.5f;
+                }
+                intrinsics.fx = rect_params[0];
+                intrinsics.fy = rect_params[1];
+                intrinsics.ppx = rect_params[2];
+                intrinsics.ppy = rect_params[3];
+                intrinsics.model = RS2_DISTORTION_BROWN_CONRADY;
+                memset(intrinsics.coeffs, 0, sizeof(intrinsics.coeffs));  // All coefficients are zeroed since rectified depth is defined as CS origin
+
+                // In case of the special 848x100 resolution adjust the intrinsics 
+                if (width == 848 && height == 100)
+                {
+                    intrinsics.height = 100;
+                    if (fps <= 90)
+                        intrinsics.ppy -= 190;
+                }
+
+                return intrinsics;
+            }
+            else
+            {
+                // Intrinsics not found in the calibration table - use the generic calculation
+                ds5_rect_resolutions resolution = res_1920_1080;
```

Comment:
The generic part assumes the sensor uses FullHd for calibration with 16/9 aspect. But for other sensors, e.g. AWGC 1MP with 1280X800 16/10 aspect ratio it will produce distortions during uv mapping, for instance.

Reply:
We can add the code to validate entry for res_1920_1080 and throw exception if the entry is not used for calibration or not valid.